src/mono/mono/mini/intrinsics.c

```diff
@@ -488,6 +488,125 @@ emit_unsafe_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignatu
 		EMIT_NEW_BIALU (cfg, ins, OP_COMPARE, -1, args [0]->dreg, args [1]->dreg);
 		EMIT_NEW_UNALU (cfg, ins, OP_PCEQ, dreg, -1);
 		return ins;
+	} else if (!strcmp (cmethod->name, "BitCast")) {
+		g_assert (ctx);
+		g_assert (ctx->method_inst);
+		g_assert (ctx->method_inst->type_argc == 2);
+		g_assert (fsig->param_count == 1);
+
+		MonoType *tfrom = ctx->method_inst->type_argv [0];
+		if (mini_is_gsharedvt_variable_type (tfrom)) {
+			// FIXME: Handle gsharevt
+			return NULL;
```

Comment:
I'm not actually convinced we need to handle it.  gsharedvt is a slow fallback strategy, we should take a look at the generated code and possibly just leave it to the inliner and optimizations.

Reply:
Good to know, have removed the `// FIXME` and explicitly called out that we don't need to handle it